.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,441 @@
+          python ci/run_tests/run_tests.py --interface en0 --use-sudo
+
+      - name: Test Examples
+        if: ${{ matrix.arch == 'x86_64' }}
+        run: |
+          cd Tests/ExamplesTest
+          python -m pip install -U pip
+          python -m pip install -r requirements.txt
+          python -m pytest --interface en0 --use-sudo --root-path=../../Dist/examples_bin
+
+      - name: Check installation
+        run: cmake --install $BUILD_DIR
+
+      - name: Build Tutorials
+        run: |
+          mkdir -p build_examples
+          cmake -DPCAPPP_BUILD_TUTORIALS=ON -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S Examples -B build_examples
+          cmake --build build_examples -j
+
+      - name: Test Tutorials
+        if: ${{ matrix.arch == 'x86_64' }}
+        run: |
+          # We need to execute the HelloWorld in the same folder as the PCAP file
+          cp build_examples/Tutorials/Tutorial-HelloWorld/Tutorial-HelloWorld Examples/Tutorials/Tutorial-HelloWorld/
+          cd Examples/Tutorials/Tutorial-HelloWorld && ./Tutorial-HelloWorld
+
+  mingw-w64:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        include:
+          - env: i686
+            sys: mingw32
+          - env: x86_64
+            sys: mingw64
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.sys}}
+          install: >-
+            git
+            mingw-w64-${{matrix.env}}-cmake
+            mingw-w64-${{matrix.env}}-gcc
+            mingw-w64-${{matrix.env}}-make
+
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.8.x"
+
+      - name: Install NPcap
+        env:
+          NPCAP_USERNAME: ${{ secrets.NPCAP_USERNAME }}
+          NPCAP_PASSWORD: ${{ secrets.NPCAP_PASSWORD }}
+        run: |
+          ci\install_npcap.bat
+          echo "PCAP_SDK_DIR=/C/Npcap-sdk" >> $env:GITHUB_ENV
+
+      - name: Configure PcapPlusPlus
+        shell: msys2 {0}
+        run: |
+          cmake -G "MinGW Makefiles" -DPCAP_ROOT=/C/Npcap-sdk -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
+
+      - name: Build PcapPlusPlus
+        shell: msys2 {0}
+        run: cmake --build $BUILD_DIR
+
+      - name: Install tcpreplay
+        run: ci\install_tcpreplay.bat
+
+      - name: Test PcapPlusPlus
+        run: |
+          python -m pip install -r ci\run_tests\requirements.txt
+          python ci\run_tests\run_tests_windows.py
+
+      - name: Test Examples
+        run: |
+          cd Tests\ExamplesTest
+          python -m pip install -r requirements.txt
+          python -m pytest --root-path=../../Dist/examples_bin
+
+  visual-studio:
+    strategy:
+      matrix:
+        os: [windows-2019, windows-2022]
+        pcap_lib: [npcap, winpcap]
+        arch: [Win32, x64]
```

Comment:
I'm not sure we need to run all of these 8 options, we currently run 4...
Maybe we can do:
- VS 2019, x64, Release, Npcap
- VS 2019, Win32, Debug, WinPcap
- VS 2022, x64, Debug, Npcap
- VS 2022, Win32, Release, WinPcap